art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
+
+func (t *ArtTree) EachChanResult() chan Result {
+	return t.EachChanResultFrom(t.root)
+}
+
+func (t *ArtTree) EachChanResultFrom(start *ArtNode) chan Result {
+	outChan := make(chan Result)
+	go func() {
+		if start != nil {
+			for n := range t.EachChanFrom(start) {
+				if n.IsLeaf() {
+					outChan <- Result{string(n.key), n.value}
+				}
+			}
+		}
+		close(outChan)
+	}()
+	return outChan
+}
```

Comment:
> if you were going to collect the results into an array, then this would be overkill. 

I'd like to know a little bit more why you think this might be overkill.  It's my thought that clients will want to get some immediate feedback as to how many results they have at a particular subtree.  I would argue that when using graphs, trees, or other data structures, you typically want immediate feedback, like so:

```
results := t.PrefixSearch("abc")
fmt.Printf(len(results))
// Perform Binary Search on results / reduce them / filter them as needed
// Display to the user how many matches the current prefix has
```

Asynchronous processing might not be the most beneficial in these cases, mainly because it pushes the responsibility on the client to tally up results, wait until the channel returns a zero value so it knows that it's closed, and effectively write some boiler plate code that requires concurrent mechanisms to manage all of this.

For example, if I was to use this as a client of `go-art`:

```
outChan := t.EachResultFrom(t.Root)

for {
  res := <- outChan
  if res == Result{} {
    break
  }

  // do my operation with res
}

// At this point I know have all my results
```

Overall, I'm not 100% convinced that this is the direction the library should go in, mainly because I want to keep the interface as simple as possible for the consumer.  I don't see a clear reason why we need asynchronous processing just yet, but I appreciate your take on it!  

====

tl;dr: Using channels is a neat concept, but I think it could lead to code bloat on the client's side, and I'd rather just return an array of results for the time being.  It's less concurrent management of resources, and it provides immediate feedback.  Given the speed optimizations of ART trees in general, I'm not sure we absolutely need concurrent processing of results at this time.